external_parser/example_joiner.cc

```diff
@@ -140,20 +141,25 @@ void example_joiner::return_example_f(void *vw, example *ex) {
   ((example_joiner *)vw)->return_example(ex);
 }
 
-int example_joiner::process_event(const v2::JoinedEvent &joined_event) {
+bool example_joiner::process_event(const v2::JoinedEvent &joined_event) {
```

Comment:
Don't we need to surface rich info about the errors so the top level caller knows why it failed?

Reply:
My approach is to keep things as local as possible, logging where appropriate and handling errors in a logical way and at the lowest level possible.

We are lucky here in that if something fails in a lower level call we need to either clean and continue or just continue to the next payload